x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -185,10 +185,15 @@ export function getWaterfallItems(
   childrenByParentId: IWaterfallGroup,
   entryTransactionItem: IWaterfallItem
 ) {
+  const referenceSet = new Set();
```

Comment:
I'm normally not a fan of code comments - but maybe one is in order here?

Reply:
i renamed the var to `visitedWaterfallItemSet` to make it more clean instead of having to use a comment